### ISSUE TITLE
Inconsistent keyboard behaviour when interacting with panel

### ISSUE THREAD

**Issue by JoelHaslam** [grafana/grafana#47005]
Grafana has recently added the ability to navigate through panels in a dashboard using the keyboard, which is amazing! However, I have noticed that this behaviour can be somewhat inconsistent.

I'm aware that this is a relatively new feature, and is a part of https://github.com/grafana/grafana/issues/36776. I'm not sure if this is covered in that card.

**Steps to reproduce**:
Use the keyboard to navigate to panels in your Grafana dashboard. Attempt to edit or view the panels currently selected by keyboard focus by pressing the 'e' or 'v' keys.

**Actual Result**:
The result I'm seeing is varying a lot. Sometimes nothing happens, other times it loads a panel that I don't currently have keyboard access to.

In the video below, you can see me using my keyboard (by pressing 'tab') to navigate to the panels, with keyboard focus on each panel I am pressing both 'e' and 'v', with nothing happening. Around the 20 second mark, I use my mouse pointer to click on the `status history with time series` panel, which then allows me to use 'e' and 'v' for that panel. However, if I change my keyboard focus to one of the other panels, I am still redirected to the `status history with time series` panel, regardless of where my keyboard focus is.

https://user-images.githubusercontent.com/63413067/160582407-2ab89360-4fda-479a-9d70-a7e28abe2a54.mp4

**Expected Result**:
By pressing 'e' or 'v' I'd be able to edit or view the panel which my keyboard is currently focussed on.

**Relevant WCAG Criteria**:
 [2.1.1 keyboard](https://www.w3.org/TR/WCAG21/#keyboard)

**Environment**:
I am using https://play.grafana.org/ (on 29/03/2022)
- Grafana version: 8.4.4 (fcfb01fae)
- User OS & Browser: Windows 11 + Microsoft Edge

**Comment by mdbridge** [comment]
as best I can tell, it's not actually possible to to use a keyboard to focus panels.    You can't really even focus using the mouse.

what's really going on as far as I can tell is that the keyboard shortcuts like 'e' operate on the panel under the mouse.  (This is not the same as focusing -- focus remains even when you remove the mouse)

**Comment by mdbridge** [comment]
FYI, this interface choice is extremely confusing as you can see from some of these reports.  People think they have selected a panel in some method but actually it's just the accident that there mouse is over it that makes it work.

**Comment by tskarhed** [comment]
Reverting the "solution", as a bigger rewrite is needed. @joshhunt provides a good idea for it:

>The problem here is that there's a conflict between tabbing through panels, and hovering. The main problem I see is that we query the :hovered panel, which isn't enough to give us the full context of what the user has done.
Here's what I think we should do.
We keep track of the panel that has attention in a new piece of dashboard-level state(?)
attention  only ever points to one panel at a time. If a new panel gets attention , it overwrites the old one
Keyboard shortcuts like e  would go to the panel that has attention
As you mouse-enter a panel, it gets attention . As you mouse-leave a panel, it loses attention
Note that this is not mouse-move or any other continuous event
Also, as you tab around or navigate with a keyboard (it receives :focus). that panel receives attention .
Basically, we have a bit of state that just remembers the last panel a user moved their mouse over, or gave focus with the keyboard. There's no longer a conflict because it's just 'last event wins'.
Thoughts?

**Comment by tskarhed** [comment]
A fix is still needed for Scenes, but for some technical reasons, it will have to wait a bit, so I can make the proper changes in the Scenes library. PR: https://github.com/grafana/scenes/pull/716
